### DISCUSSION TITLE
How to diagnose notes not converted?

### DISCUSSION THREAD

**Discussion by wright-io** [akosbalasko/yarle#227] [General]
First, thank you so much for this amazing tool! I have a very large number of notes and some of the notes in some enex files don't seem to have converted (totals are different between md files and notes in Evernote). Is there a way to see logging about notes that weren't converted? I can't find anything in the output, but there is so much output, so I'm not sure if I'm missing the right keywords to search for, or if I have to set some logging options when running Yarle?

Thanks again and thanks in advance for any insight you can share!

**Comment by akosbalasko** [comment]
Hi @wright-io ! 
I'm glad to hear that you like the tool! 
We're facing two issues that used to fail the conversion: One is if the Enex file was exported by Evernote 10, or if it contains checkboxes in a mysterious constellation, unfortunately, I still had no time to investigate it deeply. 
Yarle produces all logs to the console,
 in case of any errors a line should appear in the console starting with the following : `Failed to convert note:`
or a conversion failure produced a line starting with `Could not convert ` and the name of the Enex file .

One other optional root cause of the difference between the total number of the notes and the number of the notes converted is if your config specifies to skip the webclips by setting `skipWebClips` to true, in this case the following line is produced: `Notes skipped: ` but without and further hint about the real note name. 

Cheers, 
akos

**Reply by wright-io** [reply]
Thanks for the reply! What about encrypted notes using Evernote's encryption? Does this fail, or just bring them over encrypted? I saw only 3 of those in my second run, but lost some of the output from another set up notebooks because I wasn't sending logs to a file. I'm going to run everything again. Thanks again for taking time to answer!

**Reply by akosbalasko** [reply]
Uh, yes, encryption is an open question for Yarle, I did not test it, and I'm not sure how it could be achievable since, we have to be able to decrypt the content somehow. Yes, it may be the case. 
Please keep me updated on your investigation!

**Reply by wright-io** [reply]
Hi,
Just wanted to provide an update on the encrypted notes in Evernote: 

- I verified that the encrypted notes were successfully migrated, but were encrypted, of course
- I went back to Evernote and searched for "encryption:", which shows all notes with encryption. I don't believe there's any way to get the full cypher from Evernote to decrypt these outside Evernote, but maybe someone else knows a way this is possible
- I moved all these notes to a new notebook in Evernote and then had to individually "permanently decrypt" all notes
- Then I exported this notebook as enex and ran Yarle on these (and of course this should work as normal!)

**Reply by akosbalasko** [reply]
Hi @wright-io !

Yeah, this is a valid workaround, and I'm pretty sure that the decryption cannot be solved outside from Evernote. The only way to resolve this if Yarle uses Evernote API to grab the notes directly from Evernote instead of using enex files. In this case it may has an option to get the notes in decrypted version, but in this case of course the decryption process is still done by EN. 
Thank you for the update!
